### PR TITLE
Fix scrollbar track offset

### LINKED ISSUE #106834: Scrollbar track offset does not account for padding
The track offset of the scrollbar is not calculated correctly. It only accounts for the main axis margin and does not include the main axis padding.

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      scrollBehavior: MaterialScrollBehavior().copyWith(scrollbars: false),
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.amber),
        scrollbarTheme: ScrollbarThemeData(
          thumbColor: MaterialStateProperty.all(Colors.amber),
          trackColor: MaterialStateProperty.all(Colors.green),
          thumbVisibility: MaterialStateProperty.all(true),
          trackVisibility: MaterialStateProperty.all(true),
        )
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Scrollbar(
        child: Center(
          child: CustomScrollView(
            primary: true,
            slivers: [
              const SliverAppBar(
                title: Text('Ta-Da'),
                backgroundColor: Colors.blue,
              ),
              SliverList(
                delegate: SliverChildBuilderDelegate(
                (c, i) => Container(height: 50, child: Center(child: Text('Item $i'))),
                childCount: 50,
              ),
              )
            ]
          ),
        ),
      ), 
    );
  }
}
```

flutter doctor
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, 3.1.0-0.0.pre.1384, on macOS 12.4 21F79 darwin-x64, locale en)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0-rc2)
[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio
[✓] Android Studio (version 2020.3)
[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.3)
[✓] VS Code (version 1.68.1)
[✓] Connected device (4 available)
[✓] HTTP Host Availability
```

### PR DESCRIPTION
**Pull request by Piinks**
Fixes https://github.com/flutter/flutter/issues/106834

This accounts for the padding when calculating the offset of the track.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/widgets/scrollbar_test.dart [hunk 1]
```diff
@@ -2565,4 +2565,35 @@ void main() {
...
+        textDirection: TextDirection.ltr,
+        child: MediaQuery(
+          data: const MediaQueryData(
+            padding: EdgeInsets.all(50.0),
+          ),
+          child: RawScrollbar(
+            controller: scrollController,
+            minThumbLength: 21,
+            minOverscrollLength: 8,
+            isAlwaysShown: true,
```
Comment (xu-baolin) [inline packages/flutter/test/widgets/scrollbar_test.dart:2582]: nit: Use thumbVisibility instea.

File: packages/flutter/test/widgets/scrollbar_test.dart [hunk 2]
```diff
@@ -2565,4 +2565,35 @@ void main() {
 
     // Go without throw.
   });
+
+  testWidgets('Track offset respects padding', (WidgetTester tester) async {
```
Comment (xu-baolin) [inline packages/flutter/test/widgets/scrollbar_test.dart:2569]: // Regression test for https://github.com/flutter/flutter/issues/106834

**Review by xu-baolin — APPROVED** [review]
LGTM, with small nits.

File: packages/flutter/test/widgets/scrollbar_test.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter/test/widgets/scrollbar_test.dart:2582]: Ah thank you! I came across this while working on another proposal and just copied another test in a rush. 🙃

**Pull request merged**